homework/tests/example.spec.js

```diff
@@ -14,18 +15,184 @@ const data = [
+    })
+  });
+
+  test.describe(version + ': Add', () => {
+    test('Multiplying 2 and 3 resultns in 6', async ({ page }) => {
+      let calculatorPage = new CalculatorPage(page);
+
+      await calculatorPage.navigate();
+      await calculatorPage.getVersion(version);
+      await calculatorPage.click('#number1Field');
+      await calculatorPage.fillField('#number1Field', '2');
+      await calculatorPage.click('#number2Field');
+      await calculatorPage.fillField('#number2Field', '3');
+      await calculatorPage.selectOption('select[name="selectOperation"]', '2');
+      await calculatorPage.click('#calculateButton');
+
+      await expect(page.locator('#numberAnswerField')).toHaveValue('6');
+
+    })
+  });
+
+  test.describe(version + ': Add', () => {
+    test('Dividing 6 out of 2 resultns in 3', async ({ page }) => {
+      let calculatorPage = new CalculatorPage(page);
+
+      await calculatorPage.navigate();
+      await calculatorPage.getVersion(version);
+      await calculatorPage.click('#number1Field');
+      await calculatorPage.fillField('#number1Field', '6');
+      await calculatorPage.click('#number2Field');
+      await calculatorPage.fillField('#number2Field', '2');
+      await calculatorPage.selectOption('select[name="selectOperation"]', '3');
+      await calculatorPage.click('#calculateButton');
+
+      await expect(page.locator('#numberAnswerField')).toHaveValue('3');
+    })
+  });
+
+  test.describe(version + ': Add', () => {
+    test('Dividing 2.2 out of 1.5 resultns in 1.4666666666', async ({ page }) => {
+      let calculatorPage = new CalculatorPage(page);
+
+      await calculatorPage.navigate();
+      await calculatorPage.getVersion(version);
+      await calculatorPage.click('#number1Field');
+      await calculatorPage.fillField('#number1Field', '2.2');
+      await calculatorPage.click('#number2Field');
+      await calculatorPage.fillField('#number2Field', '1.5');
+      await calculatorPage.selectOption('select[name="selectOperation"]', '3');
+      await calculatorPage.click('#calculateButton');
+
+      await expect(page.locator('#numberAnswerField')).toHaveValue('1.46666666');
+    })
+  });
+
+  test.describe(version + ': Add', () => {
+    test('Dividing empty field results in error', async ({ page }) => {
+      let calculatorPage = new CalculatorPage(page);
+
+      await calculatorPage.navigate();
+      await calculatorPage.getVersion(version);
+      await calculatorPage.click('#number1Field');
+      await calculatorPage.fillField('#number1Field', '');
+      await calculatorPage.click('#number2Field');
+      await calculatorPage.fillField('#number2Field', '');
+      await calculatorPage.selectOption('select[name="selectOperation"]', '3');
+      await calculatorPage.click('#calculateButton');
+
+      let errorMessage = await page.$('#errorMsgField');
+
+      expect(await errorMessage?.textContent()).toEqual('Divide by zero error!');
```

Comment:
defining error message location separately and using this plus not null validations to check the message text seems a bit overcomplicated and makes reading code more difficult, why did you do it this way instead of having only one line for assertion n which you would get the message?

Reply:
It is the only way it worked for me.
Should it look like this:
expect(await page.$('#errorMsgField')?.textContent()).toEqual('Divide by zero error!'); ?
In this way, I can't access textContent().